packages/shared/src/components/modals/post/boost/BoostPostModal.tsx

```diff
@@ -0,0 +1,281 @@
+  const [coresPerDay, setCoresPerDay] = React.useState(5000);
+  const [totalDays, setTotalDays] = React.useState(7);
+  const [queryProps, setQueryProps] = React.useState({
+    coresPerDay,
+    totalDays,
+  });
+  const { data } = useQuery<{ min: number; max: number }>({
+    queryKey: generateQueryKey(
+      RequestKey.PostBoostReach,
+      user,
+      post.id,
+      queryProps.coresPerDay,
+      queryProps.totalDays,
+    ),
+  });
+  const client = useQueryClient();
+  const [debounceSet] = useDebounceFn(setQueryProps, 220);
+  const totalSpendInt = coresPerDay * totalDays;
+  const totalSpend = largeNumberFormat(totalSpendInt);
+  const { mutateAsync: onBoost } = useMutation<{
+    startPostBoost: TransactionCreated;
+  }>({
+    onSuccess: (result) => {
+      setActiveScreen(SCREENS.SUCCESS);
+      const balance = result?.startPostBoost?.balance;
+
+      if (!isNullOrUndefined(balance)) {
+        updateUser({ ...user, balance });
+      }
+
+      // invalidate wallet queries
+      client.invalidateQueries({
+        queryKey: generateQueryKey(RequestKey.Transactions, user),
+        exact: false,
+      });
+    },
+  });
+  const image = usePostImage(post);
+
+  const onButtonClick = () => {
+    if (user.balance.amount < totalSpendInt) {
+      return setActiveScreen(SCREENS.BUY_CORES);
+    }
+
+    return onBoost();
+  };
+
+  if (activeScreen === SCREENS.BUY_CORES) {
+    return (
+      <BuyCoresModal
+        isOpen
+        product={null}
+        onCompletion={() => setActiveScreen(SCREENS.FORM)}
+        onRequestClose={() => setActiveScreen(SCREENS.FORM)}
+        origin={Origin.BoostPost}
+      />
+    );
+  }
+
+  if (activeScreen === SCREENS.SUCCESS) {
+    return (
+      <Modal
+        {...props}
+        isOpen
+        kind={Modal.Kind.FixedCenter}
+        size={Modal.Size.Small}
+        isDrawerOnMobile
+      >
+        <Modal.Body className="flex flex-col">
+          <Image src={postBoostSuccessCover} />
+          <div className="flex flex-col gap-2">
+            <Typography type={TypographyType.Title2}>
+              Post boosted successfully!
+            </Typography>
+            <Typography
+              type={TypographyType.Body}
+              color={TypographyColor.Tertiary}
+            >
+              Your post is now being promoted and will start reaching more
+              developers shortly. You can track its performance anytime from the
+              ads dashboard.
+            </Typography>
+          </div>
+          <Button
+            variant={ButtonVariant.Primary}
+            className="w-full"
+            type="button"
+            onClick={() => setActiveScreen(SCREENS.DASHBOARD)}
+          >
+            Ads dashboard
+          </Button>
+        </Modal.Body>
+      </Modal>
+    );
+  }
+
+  if (activeScreen === SCREENS.DASHBOARD) {
+    return <AdsDashboardModal isOpen />;
+  }
+
+  return (
+    <Modal
```

Comment:
Guess ideally we move some of these inline modals to their own functions/files? but can be done seperate

Reply:
Agreed. Much cleaner that way. I'll move it 🙏